src/Stratis.Bitcoin.IntegrationTests/EnvironmentMockUpHelpers/NodeRunners.cs

```diff
@@ -135,65 +154,112 @@ public static IFullNode BuildStakingNode(string dir, bool staking = true)
         }
 
-        public bool IsDisposed
+        public override FullNode OnBuild(NodeSettings nodeSettings)
         {
-            get { return this.FullNode.State == FullNodeState.Disposed; }
+            FullNode fullNode;
+
+            if (this.Callback != null)
+                fullNode = base.BuildFromCallBack(nodeSettings);
+            else
+            {
+                IFullNodeBuilder builder = null;
+                builder = Build(nodeSettings).UsePowConsensus().AddMining().MockIBD();
+                if (this.SkipRules)
+                    builder.MockRules(new PowTestRuleRegistration());
+                fullNode = (FullNode)builder.Build();
+            }
+
+            return fullNode;
         }
+    }
 
-        public void Kill()
+    public sealed class StratisPowRunner : NodeRunner
+    {
+        public StratisPowRunner(bool skipRules, Action<IFullNodeBuilder> callback = null)
+            : base(skipRules, callback)
         {
-            this.FullNode?.Dispose();
         }
 
-        public void Start(string dataDir)
+        public override void OnStart(string dataDir)
         {
-            NodeSettings nodeSettings = new NodeSettings(args:new string[] { "-conf=bitcoin.conf", "-datadir=" + dataDir }, loadConfiguration:false);
+            base.Start(new NodeSettings(Network.StratisRegTest, ProtocolVersion.ALT_PROTOCOL_VERSION, args: new string[] { "-conf=stratis.conf", "-datadir=" + dataDir }, loadConfiguration: false));
+        }
 
-            var node = BuildFullNode(nodeSettings, this.callback);
+        public override FullNode OnBuild(NodeSettings nodeSettings)
+        {
+            FullNode fullNode;
 
-            this.FullNode = node;
-            this.FullNode.Start();
+            if (this.Callback != null)
+                fullNode = BuildFromCallBack(nodeSettings);
+            else
+            {
+                IFullNodeBuilder builder = null;
+                builder = Build(nodeSettings).UsePosConsensus().AddMining().MockIBD();
+                if (this.SkipRules)
+                    builder.MockRules(new PosTestRuleRegistration());
+                fullNode = (FullNode)builder.Build();
+            }
+
+            return fullNode;
         }
+    }
 
-        public static FullNode BuildFullNode(NodeSettings args, Action<IFullNodeBuilder> callback = null)
+    public sealed class PowTestRuleRegistration : IRuleRegistration
+    {
+        public IEnumerable<ConsensusRule> GetRules()
         {
-            FullNode node;
+            return new List<ConsensusRule>() { new PowTestRule() };
+        }
+    }
 
-            if (callback != null)
-            {
-                var builder = new FullNodeBuilder().UseNodeSettings(args);
+    public sealed class PosTestRuleRegistration : IRuleRegistration
+    {
+        public IEnumerable<ConsensusRule> GetRules()
+        {
+            return new List<ConsensusRule>() { new PosTestRule() };
```

Comment:
There is only one rule added, so this means all other rules are not run then the measurement of the test is not so accurate.